pkg/podgrouper/podgrouper/plugins/defaultgrouper/default_grouper.go

```diff
@@ -157,42 +162,84 @@ func (dg *DefaultGrouper) calculateQueueName(topOwner *unstructured.Unstructured
 
 func (dg *DefaultGrouper) CalcPodGroupPriorityClass(topOwner *unstructured.Unstructured, pod *v1.Pod,
 	defaultPriorityClassForJob string) string {
+	priorityClassName, _ := dg.calcPriorityClassWithDefaults(topOwner, pod, defaultPriorityClassForJob)
+	return priorityClassName
+}
+
+// calcPriorityClassWithDefaults - resolves priority class using:
+// 1) explicit labels (top owner/pod), if valid
+// 2) defaults from ConfigMap (returned to allow reuse by caller)
+// 3) final fallback to defaultPriorityClassForJob
+// Returns the resolved priority class name and the defaults mapping used (if any).
+func (dg *DefaultGrouper) calcPriorityClassWithDefaults(topOwner *unstructured.Unstructured, pod *v1.Pod,
+	defaultPriorityClassForJob string) (string, map[string]workloadTypePriorityConfig) {
 	priorityClassName := dg.calcPodGroupPriorityClass(topOwner, pod)
 	if dg.validatePriorityClassExists(priorityClassName) {
-		return priorityClassName
+		return priorityClassName, nil
 	}
-
 	if priorityClassName != "" {
 		logger.V(1).Info("priorityClassName from pod or owner labels is not valid, falling back to default",
 			"priorityClassName", priorityClassName, "topOwner", topOwner.GetName(), "pod", pod.GetName())
 	}
 
 	groupKind := topOwner.GroupVersionKind().GroupKind()
-	priorityClassName = dg.getDefaultPriorityClassNameForKind(&groupKind)
-	if dg.validatePriorityClassExists(priorityClassName) {
-		return priorityClassName
+	defaultConfigs, err := dg.getDefaultConfigsPerTypeMapping()
+	if err == nil {
+		priorityClassName = dg.getDefaultPriorityClassNameForKind(&groupKind, defaultConfigs)
+		if dg.validatePriorityClassExists(priorityClassName) {
+			return priorityClassName, defaultConfigs
+		}
+	} else {
+		logger.Error(err, "Unable to get default values mapping for priority class", "topOwner", topOwner.GetName(), "pod", pod.GetName())
 	}
 
 	logger.V(1).Info("No default priority class found for group kind, using default fallback",
 		"groupKind", groupKind.String(), "defaultFallback", defaultPriorityClassForJob)
-	return defaultPriorityClassForJob
+	return defaultPriorityClassForJob, defaultConfigs
 }
 
-func (dg *DefaultGrouper) calcPodGroupPreemptibility(topOwner *unstructured.Unstructured, pod *v1.Pod) v2alpha2.Preemptibility {
-	if preemptibility, found := topOwner.GetLabels()[constants.PreemptibilityLabelKey]; found {
-		if preemptibility, err := v2alpha2.ParsePreemptibility(preemptibility); err == nil {
+// calcPodGroupPreemptibilityWithDefaults - resolves preemptibility using:
+// 1) explicit labels (top owner/pod)
+// 2) provided defaults (if any)
+// 3) defaults from ConfigMap
```

Comment:
I think the implementation has 2 and 3 swapped.
Maybe the comment is not a must have in this code.